.github/workflows/update-documentation.yml

```
@@ -0,0 +1,135 @@
+            NUM_CHANGED_CHARTS=$((NUM_CHANGED_CHARTS + 1))
+            echo "CHANGED_CHART=\"$chart\"" >> "$GITHUB_ENV"
+          done
+          if [ "$NUM_CHANGED_CHARTS" -gt 1 ]; then
+            echo "Multiple charts have changed, only update one chart per PR"
+            exit 1
+          fi
+          if git diff --name-only HEAD~1 | grep -q "$CHANGED_CHART/values.yaml"; then
+            echo "$CHANGED_CHART: No values.yaml changes detected"
+          fi
+          echo "$CHANGED_CHART: values.yaml changes detected"
+          echo "VALUES_CHANGED=true" >> "$GITHUB_ENV"
+
+  update-documentation:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs: gather-information
+    if: ${{ needs.gather-information.outputs.failed != 'true' }}
+    env:
+      PR_TITLE: ${{ github.event.pull_request.title }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          token: ${{ secrets.CONTANE_BOT_TOKEN }}
+
+      - name: Configure Git
+        run: |
+          git config user.name "contane-bot"
+          git config user.email "160241315+contane-bot@users.noreply.github.com"
+
+      - name: Install readme-generator-for-helm
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        run: npm install -g @bitnami/readme-generator-for-helm@2.6.1
+
+      - name: Update README.md
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        shell: bash
+        run: |
+          cd charts || exit 1
+          echo "Updating README.md for $CHANGED_CHART"
+          readme-generator --values "$CHANGED_CHART/values.yaml" --readme "$CHANGED_CHART/README.md"
+
+      - name: Update changelog.md
+        shell: bash
+        run: |
+          echo "Generating changelog for $CHANGED_CHART"
+          npx semantic-release --dry-run --no-ci --plugins @semantic-release/release-notes-generator > "$CHANGED_CHART/release-notes.md"
+          # Extract relevant part
+          sed -n '/### \[/{:a;n;/### \[/{p;q};p;ba}' "$CHANGED_CHART/release-notes.md" > "$CHANGED_CHART/CHANGELOG.md"
+          rm "$CHANGED_CHART/release-notes.md"
+
+      - name: Install changelog-generator
+        run: npm install -g github-changelog-generator
+
+      - name: Update ArtifactHub change annotations
+        shell: bash
+        run: |
+          # Parse PR title to create changelog entries
+          changes=""
+          if [[ $PR_TITLE == *"feat:"* ]]; then
+            changes="$changes"$'\n'  - kind: feature
+            changes="$changes"$'\n'    description: "${PR_TITLE#*: }"
+          elif [[ "$PR_TITLE" == *"fix:"* ]]; then
+            changes="$changes"$'\n'  - kind: bugfix
+            changes="$changes"$'\n'    description: "${PR_TITLE#*: }"
+          fi
```

Wouldn't this work? (Besides generating invalid YAML if the PR title contains quotes or backslashes)

```suggestion
          if [[ "$PR_TITLE" == *"feat:"* ]]; then
            changes="  - kind: feature\n    description: \"${PR_TITLE#*: }\""
          elif [[ "$PR_TITLE" == *"fix:"* ]]; then
            changes="  - kind: bugfix\n    description: \"${PR_TITLE#*: }\""
          fi
```